erpnext/schools/doctype/student_group/student_group.js

```diff
@@ -1,19 +1,37 @@
 cur_frm.add_fetch("student", "title", "student_name");
 
 frappe.ui.form.on("Student Group", {
+	onload: function(frm) {
+		frm.set_query("academic_term", function() {
+			return {
+				"filters": {
+					"academic_year": (frm.doc.academic_year)
+				}
+			};
+		});
+	},
+
+	before_save: function(frm) {
+		var name;
+		if (frm.doc.group_based_on == "Course") {
+			name = "Course-" + frm.doc.course + "-" + (frm.doc.academic_term?frm.doc.academic_term:frm.doc.academic_year);
+		} else if (frm.doc.group_based_on == "Batch") {
+			name = "Batch-" + frm.doc.program + "-" + frm.doc.batch + "-" + (frm.doc.academic_term?frm.doc.academic_term:frm.doc.academic_year); 
+		} else if (frm.doc.group_based_on == "Activity") {
+			name = "Activity" + "-" + (frm.doc.academic_term?frm.doc.academic_term:frm.doc.academic_year);
+		}
+		if (!frm.doc.__unnamed) {
+			frm.doc.__newname = name;
```

Comment:
Instead of prompt, add a field "Student Group Name" and set the suggested name there. It will help for naming while pushing student group data via API.